### DISCUSSION TITLE
Define default value for an enum

### DISCUSSION THREAD

**Discussion by srika4** [json-schema-org/community#519] [Q&A]
Hi. Looking for an example to define a "default value" with an enum.

```jsonc
{
  "properties": {
    "Param": {
      "const": "p1", // not working
      "default": "p1", // is this correct and can validate?
      "enum": [
        "p1",
        "p3"
      ]
    }
  }
}
```

**Comment by gregsdennis** [comment]
> `"const": "p1", // not working`

Can you show instances where this isn't working?

> `"default": "p1", // is this correct and can validate?`

`default` is just an annotation.  It doesn't validate, and it doesn't provide a value to be validated if the instance doesn't have that property.  It's just there as information for the consumer.

> `"const": ..., "enum": ... "

Both of these keywords are intended to be used independently.  `const` requires that the instance is _exactly_ that value, and `enum` can be considered a `oneOf` with a bunch of `const`s in it.

```json
{ "enum": [ 1, 2, 3 ] }
```

is the same as 

```json
{
  "oneOf": [
    { "const": 1 },
    { "const": 2 },
    { "const": 3 }
  ]
}
